chain/network/src/routing.rs

```diff
@@ -89,6 +95,14 @@ pub struct Edge {
     removal_info: Option<(bool, Signature)>,
 }
 
+impl Hash for Edge {
```

Comment:
This feels wrong -- if we impleent Hash manually, we should not derive `Eq`.

Reply:
We can't derive `Hash`, because Signature doesn't implement it, we can't add it, because it's fields are imported from outside our repository.

However, Signatures are unique, there is only one possible value for them based on (peer0, peer1, nonce). However, we need then, because we can't generate them without public keys.

I'm fine implementing PartialEq manually, but there isn't a point.

Reply:
Makes sense! I guess a short comment that the generated Eq is fine wouldn't hurt though.